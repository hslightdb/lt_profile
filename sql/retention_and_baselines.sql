UPDATE profile.samples SET sample_time = now() - '4 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 1;
UPDATE profile.samples SET sample_time = now() - '3 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 2;
UPDATE profile.samples SET sample_time = now() - '2 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 3;
UPDATE profile.samples SET sample_time = now() - '1 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 4;
UPDATE profile.samples SET sample_time = now() - '23 hours'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 5;
SELECT server,result FROM profile.take_sample();
SELECT count(*) FROM profile.samples WHERE sample_time < now() - '1 days'::interval;
SELECT * FROM profile.set_server_max_sample_age('local',1);
/* Testing baseline creation */
SELECT * FROM profile.create_baseline('testline1',2,4);
SELECT * FROM profile.create_baseline('testline2',2,4);
SELECT count(*) FROM profile.baselines;
SELECT * FROM profile.keep_baseline('testline2',-1);
/* Testing baseline show */
SELECT baseline, min_sample, max_sample, keep_until_time IS NULL
FROM profile.show_baselines()
ORDER BY baseline;
/* Testing baseline deletion */
SELECT server,result FROM profile.take_sample();
SELECT count(*) FROM profile.baselines;
/* Testing samples retention override with baseline */
SELECT count(*) FROM profile.samples WHERE sample_time < now() - '1 days'::interval;
SELECT * FROM profile.drop_baseline('testline1');
/* Testing samples deletion after baseline removed */
SELECT server,result FROM profile.take_sample();
SELECT count(*) FROM profile.samples WHERE sample_time < now() - '1 days'::interval;
