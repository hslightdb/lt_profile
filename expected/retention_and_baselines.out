UPDATE profile.samples SET sample_time = now() - '4 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 1;
UPDATE profile.samples SET sample_time = now() - '3 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 2;
UPDATE profile.samples SET sample_time = now() - '2 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 3;
UPDATE profile.samples SET sample_time = now() - '1 days'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 4;
UPDATE profile.samples SET sample_time = now() - '23 hours'::interval - '10 minutes'::interval WHERE server_id = 1 AND sample_id = 5;
SELECT server,result FROM profile.take_sample();
 server | result 
--------+--------
 local  | OK
(1 row)

SELECT count(*) FROM profile.samples WHERE sample_time < now() - '1 days'::interval;
 count 
-------
     4
(1 row)

SELECT * FROM profile.set_server_max_sample_age('local',1);
 set_server_max_sample_age 
---------------------------
                         1
(1 row)

/* Testing baseline creation */
SELECT * FROM profile.create_baseline('testline1',2,4);
 create_baseline 
-----------------
               1
(1 row)

SELECT * FROM profile.create_baseline('testline2',2,4);
 create_baseline 
-----------------
               2
(1 row)

SELECT count(*) FROM profile.baselines;
 count 
-------
     2
(1 row)

SELECT * FROM profile.keep_baseline('testline2',-1);
 keep_baseline 
---------------
             1
(1 row)

/* Testing baseline show */
SELECT baseline, min_sample, max_sample, keep_until_time IS NULL
FROM profile.show_baselines()
ORDER BY baseline;
 baseline  | min_sample | max_sample | ?column? 
-----------+------------+------------+----------
 testline1 |          2 |          4 | t
 testline2 |          2 |          4 | f
(2 rows)

/* Testing baseline deletion */
SELECT server,result FROM profile.take_sample();
 server | result 
--------+--------
 local  | OK
(1 row)

SELECT count(*) FROM profile.baselines;
 count 
-------
     1
(1 row)

/* Testing samples retention override with baseline */
SELECT count(*) FROM profile.samples WHERE sample_time < now() - '1 days'::interval;
 count 
-------
     3
(1 row)

SELECT * FROM profile.drop_baseline('testline1');
 drop_baseline 
---------------
             1
(1 row)

/* Testing samples deletion after baseline removed */
SELECT server,result FROM profile.take_sample();
 server | result 
--------+--------
 local  | OK
(1 row)

SELECT count(*) FROM profile.samples WHERE sample_time < now() - '1 days'::interval;
 count 
-------
     0
(1 row)

